app/src/apps/Examinations/Lab.svelte

```diff
@@ -45,8 +45,114 @@
       on:update={updateLab}
     />
   {:else}
-    <!-- TODO: add loading animation -->
-    <div>Loading...</div>
+    <div class="ball-grid-beat">
+      <div />
+      <div />
+      <div />
+      <div />
+      <div />
+      <div />
+      <div />
+      <div />
+      <div />
+    </div>
+
+    <style>
+      .ball-grid-beat > div {
+        -webkit-animation-fill-mode: both;
+        -webkit-animation-iteration-count: infinite;
```

Comment:
Instead of putting this directly into this div, it would be better if we created a reusable component in the [components folder](https://github.com/CarletonComputerScienceSociety/discretemath.ca/blob/18d17e7ea87d57d445c875220d7073ddd74c96fd/app/src/components) and imported it here.